### ISSUE TITLE
undefined method `before_filter'

### ISSUE THREAD

**Issue by rmkanda** [zorab47/active_admin-duplicatable#6]
undefined method `before_filter' error

/usr/local/lib/ruby/gems/2.5.0/gems/active_admin-duplicatable-0.3.0/lib/active_admin/duplicatable.rb:49:in `block in enable_resource_duplication_via_form': undefined method `before_filter' for Admin::OfferingsController:Class (NoMethodError)
Did you mean?  before_build

**Environment:**
Ruby 2.5.1p57
Rails 5.2.0
Activeadmin 1.3.0

**Full Trace:**

```
> $ rails s
> => Booting Puma
> => Rails 5.2.0 application starting in development 
> => Run `rails server -h` for more startup options
> Exiting
> Traceback (most recent call last):
>         112: from bin/rails:3:in `<main>'
>         111: from bin/rails:3:in `load'
>         110: from /Users/rmkanda/Capable/github-git/capable-admin/bin/spring:15:in `<top (required)>'
>         109: from /usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:70:in `require'
>         108: from /usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:70:in `require'
>         107: from /usr/local/lib/ruby/gems/2.5.0/gems/spring-2.0.2/lib/spring/binstub.rb:31:in `<top (required)>'
>         106: from /usr/local/lib/ruby/gems/2.5.0/gems/spring-2.0.2/lib/spring/binstub.rb:31:in `load'
>         105: from /usr/local/lib/ruby/gems/2.5.0/gems/spring-2.0.2/bin/spring:49:in `<top (required)>'
>         104: from /usr/local/lib/ruby/gems/2.5.0/gems/spring-2.0.2/lib/spring/client.rb:30:in `run'
>         103: from /usr/local/lib/ruby/gems/2.5.0/gems/spring-2.0.2/lib/spring/client/command.rb:7:in `call'
>         102: from /usr/local/lib/ruby/gems/2.5.0/gems/spring-2.0.2/lib/spring/client/rails.rb:28:in `call'
>         101: from /usr/local/lib/ruby/gems/2.5.0/gems/spring-2.0.2/lib/spring/client/rails.rb:28:in `load'
>         100: from /Users/rmkanda/Capable/github-git/capable-admin/bin/rails:9:in `<top (required)>'
>          99: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `require'
>          98: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:249:in `load_dependency'
>          97: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `block in require'
>          96: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
>          95: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
>          94: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
>          93: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
>          92: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
>          91: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/commands.rb:18:in `<main>'
>          90: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/command.rb:46:in `invoke'
>          89: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/command/base.rb:65:in `perform'
>          88: from /usr/local/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
>          87: from /usr/local/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
>          86: from /usr/local/lib/ruby/gems/2.5.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
>          85: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:142:in `perform'
>          84: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:142:in `tap'
>          83: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:147:in `block in perform'
>          82: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:51:in `start'
>          81: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:89:in `log_to_stdout'
>          80: from /usr/local/lib/ruby/gems/2.5.0/gems/rack-2.0.5/lib/rack/server.rb:354:in `wrapped_app'
>          79: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/commands/server/server_command.rb:27:in `app'
>          78: from /usr/local/lib/ruby/gems/2.5.0/gems/rack-2.0.5/lib/rack/server.rb:219:in `app'
>          77: from /usr/local/lib/ruby/gems/2.5.0/gems/rack-2.0.5/lib/rack/server.rb:319:in `build_app_and_options_from_config'
>          76: from /usr/local/lib/ruby/gems/2.5.0/gems/rack-2.0.5/lib/rack/builder.rb:40:in `parse_file'
>          75: from /usr/local/lib/ruby/gems/2.5.0/gems/rack-2.0.5/lib/rack/builder.rb:49:in `new_from_string'
>          74: from /usr/local/lib/ruby/gems/2.5.0/gems/rack-2.0.5/lib/rack/builder.rb:49:in `eval'
>          73: from config.ru:in `<main>'
>          72: from config.ru:in `new'
>          71: from /usr/local/lib/ruby/gems/2.5.0/gems/rack-2.0.5/lib/rack/builder.rb:55:in `initialize'
>          70: from /usr/local/lib/ruby/gems/2.5.0/gems/rack-2.0.5/lib/rack/builder.rb:55:in `instance_eval'
>          69: from config.ru:3:in `block in <main>'
>          68: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:44:in `require_relative'
>          67: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `require'
>          66: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:249:in `load_dependency'
>          65: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:283:in `block in require'
>          64: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
>          63: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
>          62: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
>          61: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
>          60: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
>          59: from /Users/rmkanda/Capable/github-git/capable-admin/config/environment.rb:5:in `<main>'
>          58: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/application.rb:361:in `initialize!'
>          57: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/initializable.rb:60:in `run_initializers'
>          56: from /usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/tsort.rb:205:in `tsort_each'
>          55: from /usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/tsort.rb:226:in `tsort_each'
>          54: from /usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/tsort.rb:347:in `each_strongly_connected_component'
>          53: from /usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/tsort.rb:347:in `call'
>          52: from /usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/tsort.rb:347:in `each'
>          51: from /usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/tsort.rb:349:in `block in each_strongly_connected_component'
>          50: from /usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/tsort.rb:431:in `each_strongly_connected_component_from'
>          49: from /usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
>          48: from /usr/local/Cellar/ruby/2.5.1/lib/ruby/2.5.0/tsort.rb:228:in `block in tsort_each'
>          47: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/initializable.rb:61:in `block in run_initializers'
>          46: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/initializable.rb:32:in `run'
>          45: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/initializable.rb:32:in `instance_exec'
>          44: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/application/finisher.rb:130:in `block in <module:Finisher>'
>          43: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/application/routes_reloader.rb:33:in `execute_if_updated'
>          42: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/application/routes_reloader.rb:44:in `updater'
>          41: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/file_update_checker.rb:83:in `execute'
>          40: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/application/routes_reloader.rb:43:in `block in updater'
>          39: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/application/routes_reloader.rb:20:in `reload!'
>          38: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/application/routes_reloader.rb:57:in `load_paths'
>          37: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/application/routes_reloader.rb:57:in `each'
>          36: from /usr/local/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/application/routes_reloader.rb:57:in `block in load_paths'
>          35: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:277:in `load'
>          34: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:249:in `load_dependency'
>          33: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:277:in `block in load'
>          32: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:50:in `load'
>          31: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:50:in `load'
>          30: from /Users/rmkanda/Capable/github-git/capable-admin/config/routes.rb:2:in `<main>'
>          29: from /usr/local/lib/ruby/gems/2.5.0/gems/actionpack-5.2.0/lib/action_dispatch/routing/route_set.rb:414:in `draw'
>          28: from /usr/local/lib/ruby/gems/2.5.0/gems/actionpack-5.2.0/lib/action_dispatch/routing/route_set.rb:432:in `eval_block'
>          27: from /usr/local/lib/ruby/gems/2.5.0/gems/actionpack-5.2.0/lib/action_dispatch/routing/route_set.rb:432:in `instance_exec'
>          26: from /Users/rmkanda/Capable/github-git/capable-admin/config/routes.rb:8:in `block in <main>'
>          25: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin.rb:77:in `routes'
>          24: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/application.rb:248:in `routes'
>          23: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/application.rb:226:in `load!'
>          22: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/application.rb:226:in `each'
>          21: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/application.rb:226:in `block in load!'
>          20: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/application.rb:234:in `load'
>          19: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/error.rb:41:in `capture'
>          18: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/application.rb:234:in `block in load'
>          17: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:277:in `load'
>          16: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:249:in `load_dependency'
>          15: from /usr/local/lib/ruby/gems/2.5.0/gems/activesupport-5.2.0/lib/active_support/dependencies.rb:277:in `block in load'
>          14: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:50:in `load'
>          13: from /usr/local/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:50:in `load'
>          12: from /Users/rmkanda/Capable/github-git/capable-admin/app/admin/offerings.rb:1:in `<main>'
>          11: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin.rb:73:in `register'
>          10: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/application.rb:176:in `register'
>           9: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/namespace.rb:52:in `register'
>           8: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/namespace.rb:230:in `parse_registration_block'
>           7: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/dsl.rb:15:in `run_registration_block'
>           6: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/dsl.rb:15:in `instance_exec'
>           5: from /Users/rmkanda/Capable/github-git/capable-admin/app/admin/offerings.rb:7:in `block in <main>'
>           4: from /usr/local/lib/ruby/gems/2.5.0/gems/active_admin-duplicatable-0.3.0/lib/active_admin/duplicatable.rb:29:in `duplicatable'
>           3: from /usr/local/lib/ruby/gems/2.5.0/gems/active_admin-duplicatable-0.3.0/lib/active_admin/duplicatable.rb:48:in `enable_resource_duplication_via_form'
>           2: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/dsl.rb:74:in `controller'
>           1: from /usr/local/lib/ruby/gems/2.5.0/gems/activeadmin-1.3.0/lib/active_admin/dsl.rb:74:in `class_exec'
**> /usr/local/lib/ruby/gems/2.5.0/gems/active_admin-duplicatable-0.3.0/lib/active_admin/duplicatable.rb:49:in `block in enable_resource_duplication_via_form': undefined method `before_filter' for Admin::OfferingsController:Class (NoMethodError)
> Did you mean?  before_build**
```

**Comment by rmkanda** [comment]
**before_filter** has been deprecated in Rails 5.0 and removed in 5.1.

**Comment by zorab47** [comment]
Thanks, it is time to update that to before_action.

**Comment by zorab47** [comment]
Fixed in v0.4.0 via #7
